.github/workflows/tests-npm.yml

```diff
@@ -0,0 +1,69 @@
+name: NPM Tests
+on:
+  workflow_call:
+    inputs:
+      env_vars:
+        required: false
+        type: string
+        default: '{}'
+      npm_build_command:
+        required: false
+        type: string
+        default: ''
+      pre_test_command:
+        required: false
+        type: string
+        default: ''
+      docker_compose_file:
+        required: false
+        type: string
+        default: docker-compose.yml
+      tests:
+        required: false
+        type: string
+        default: '[test:unit, test:integration]'
+jobs:
+  tests:
+    name: Run tests - ${{ matrix.command }}
+    strategy:
+      fail-fast: false
+      matrix:
+        command: ${{ github.event.inputs.tests }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set environment variables from JSON
+        env:
+          ENV_VARS_JSON: "${{ inputs.env_vars }}"
+        run: |
+          echo "Parsing JSON environment variables..."
+          echo "$ENV_VARS_JSON" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]' >> $GITHUB_ENV
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - name: Install Packages
+        run: npm ci
+      - name: Build
+        run: ${{ inputs.npm_build_command }}
+        if: inputs.npm_build_command == true
```

Comment:
If `npm_build_command` is a `string` I don't know if this will coerce correctly. Can we change to `!= ''`

Reply:
It does coerce correctly.

https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#literals

Reply:
that's about coercing literals to booleans, which yes would coerce `''`  to `false`. Importantly the section on operators https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#operators says:

> if the types do not match, GitHub coerces the type to a number. GitHub casts data types to a number using these conversions: 
> Boolean	true returns 1 false returns 0
> String	Parsed from any legal JSON number format, otherwise NaN. Note: empty string returns 0.
> 

This would imply to me in the comparrison that `true` will coerce to `1`  and the `''`  will coerce to `NaN` which are not equal. Either way I think this is unclear

Reply:
This has now been done.